### ISSUE TITLE
parsing array of fields

### ISSUE THREAD

**Issue by alfredopacino** [macek/jquery-serialize-object#83]
This is my form
```
<form id="tA">
 <select class="form-control " name="array[][id]"id="fundamentalrights" data-select="mysettings"></select>
 <input type="text" name="array[][comment]" class="form-control" />
</form>
```

$('form#tA').serializeObject() works fine and builds the obj this way
```
{
array:[
{"id":"#ID","comment":"mycomments"}
]
}

```

the problem is #ID is wrong because the select is a [select2 ](https://select2.github.io/) element.
So I have to replace the array[0][id] value with its select2 value: $(selector).select2('data');

this code obiouvsly break the array structure made by jquery-serialize-object
```
        var form = $('form#tA').serializeObject()
        $('[data-select]').each(function() {
           var name =  $(this).attr('name');
           form[name] = $(this).select2('data');
        });

```
printing this
```
{
array[][id]:#CORRECT ID VALUE
array[][comment]:"mycomments"
}
```

maybe you have a correct way to keep the array structure?
Basically I need a re-serialization after those edits

**Comment by macek** [comment]
Your question isn't very clear, but it seems like you're trying to group values together for a dynamic-length form

```html
<input name="array[][id]" value="1">
<input name="array[][comment]" value="2">
```

Will result in

```json
{
  "array": [
    {"id": "1"},
    {"comment": "2"}
  ]
}
```

If you intend to keep these values together, you have to manually specify indexes

```html
<input name="array[0][id]" value="1">
<input name="array[0][comment]" value="2">
<input name="array[1][id]" value="a">
<input name="array[]1[comment]" value="b">
```

Will result in

```json
{
  "array": [
    {
      "id": "1"
      "comment": "2"
    },
    {
      "id": "a"
      "comment": "b"
    }
  ]
}
```

**Comment by alfredopacino** [comment]
the real name isn't array :)
thanks
